File: Data Structures & Algorithms/pow-x-n/submission-7.cs
public class Solution {
    public double MyPow(double x, int n) {
        if (x == 0) {
            return 0;
        }
        if (n == 0) {
            return 1;
        }

        double res = 1;
        for (int i = 0; i < Math.Abs(n); i++) {
            res *= x;
        }
        return n >= 0 ? res : 1 / res;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-7.java
public class Solution {
    public List<List<Integer>> combinationSum(int[] nums, int target) {
        List<List<Integer>> res = new ArrayList<>();
        Arrays.sort(nums);
        
        dfs(0, new ArrayList<>(), 0, nums, target, res);
        return res;
    }

    private void dfs(int i, List<Integer> cur, int total, int[] nums, int target, List<List<Integer>> res) {
        if (total == target) {
            res.add(new ArrayList<>(cur));
            return;
        }
        
        for (int j = i; j < nums.length; j++) {
            if (total + nums[j] > target) {
                return;
            }
            cur.add(nums[j]);
            dfs(j, cur, total + nums[j], nums, target, res);
            cur.remove(cur.size() - 1);
        }
    }
}